zfit/models/dist_tfp.py

```diff
@@ -714,3 +717,115 @@ class StudentTPDFRepr(BasePDFRepr):
+        It is related to the Student's t-distribution according to https://en.wikipedia.org/wiki/Q-Gaussian_distribution#Student's_t-distribution
+        and that is how it is implemented here.
+
+        The q-Gaussian shape for 1 < q < 3 is defined as
+
+        .. math::
+
+            f(x \\mid q, \\mu, \\sigma) = \\frac{1}{\\sigma} e_q(-\\frac{(x - \\mu)^2}{2 \\sigma^2}) / C_{q}
+
+        with
+
+        .. math::
+
+            e_q(x) = \\left[1 + (1 - q) x\\right]_{+}^{\\frac{1}{1 - q}}
+
+        and the normalization over [-inf, inf] of
+
+        .. math::
+
+            C_{q} == \\frac{\\sqrt{\\pi} \\Gamma \\left(\\frac{3 - q}{2 (q - 1)}\\right)}{\\sqrt{q - 1}\\Gamma \\left(\\frac{1}{q - 1}\\right)}
+
+        The normalization changes for different normalization ranges
+
+        Args:
+            q: Shape parameter of the q-Gaussian. Must be 1 < q < 3.
+            mu: Mean of the distribution
+            sigma: Scale of the distribution
+            obs: |@doc:model.init.obs| Observables of the
+               model. This will be used as the default space of the PDF and,
+               if not given explicitly, as the normalization range.
+
+               The default space is used for example in the sample method: if no
+               sampling limits are given, the default space is used.
+
+               The observables are not equal to the domain as it does not restrict or
+               truncate the model outside this range. |@docend:model.init.obs|
+            extended: |@doc:pdf.init.extended| The overall yield of the PDF.
+               If this is parameter-like, it will be used as the yield,
+               the expected number of events, and the PDF will be extended.
+               An extended PDF has additional functionality, such as the
+               ``ext_*`` methods and the ``counts`` (for binned PDFs). |@docend:pdf.init.extended|
+            norm: |@doc:pdf.init.norm| Normalization of the PDF.
+               By default, this is the same as the default space of the PDF. |@docend:pdf.init.norm|
+            name: |@doc:model.init.name| Human-readable name
+               or label of
+               the PDF for better identification.
+               Has no programmatical functional purpose as identification. |@docend:model.init.name|
+        """
+        q, mu, sigma = self._check_input_params(q, mu, sigma)
+        if q < 1 or q > 3:
+            msg = "q < 1 or q > 3 are not supported"
+            raise ValueError(msg)
+        if q == 1:
+            msg = "q = 1 is a Gaussian, use Gauss instead."
+            raise ValueError(msg)
+        params = OrderedDict((("q", q), ("mu", mu), ("sigma", sigma)))
+
+        # https://en.wikipedia.org/wiki/Q-Gaussian_distribution
+        # relation to Student's t-distribution
+
+        # 1/(2 sigma^2) = 1 / (3 - q)
+        # 2 sigma^2 = 3 - q
+        # sigma = sqrt((3 - q)/2)
+
+        def dist_params():
```

Comment:
```suggestion
        def dist_params(q=q, mu=mu, sigma=sigma):
```
We can "capture" them: this way, the q is really q (by default, but we will never change it), i.e. by _reference_. The danger by not using this pattern is that `q` is only saved inside the function as a _lexical_, i.e. as a name.  For example, if later on in the init we would change `q` to something else, _that's_ what the function will use inside (it's gonna be a lexical lookup instead of by reference)

(yes, this isn't done with others (yet), but would be better if it were)